frontend/src/components/nlp/components/NlpSample.tsx

```diff
@@ -285,7 +285,7 @@ export default function NlpSample() {
             ) ? (
               <Button
                 variant="contained"
-                href={`${publicRuntimeConfig.apiUrl}/nlpsample/export${
+                href={`${publicRuntimeConfig.apiUrl}nlpsample/export${
```

Comment:
Thank you @kartik1112 for adressing this 🙏 

publicRuntimeConfig.apiUrl is a config value that can be passed as a env var.

This config could be missing a trailing slash. This would lead to have a link like the following : http://localhost:4000nlpsample/export/....


I suggest we add a helper function in the utils/URL.ts file : 
```ts
/**
 * Joins a base URL and a relative path to form a complete URL.
 *
 * @param baseUrl - The base URL (must include the protocol, e.g., 'http://', 'https://').
 * @param relativePath - The relative URL path to be appended to the base URL.
 * @returns The full URL as a string.
 */
function buildUrl(baseUrl: string, relativePath: string): string {
    // Check if baseUrl is a valid URL
    try {
        new URL(baseUrl);
    } catch {
        throw new Error(`Invalid base URL: ${baseUrl}`);
    }

    // Create and return the full URL
    const url = new URL(relativePath, baseUrl);
    return url.toString();
}

// Example usage:
const domain = 'https://example.com/some/path/';
const path = '../another/path';

const fullUrl = buildUrl(domain, path);

console.log(fullUrl); // Outputs: 'https://example.com/another/path'
```

If you could please search for other usages of "publicRuntimeConfig.apiUrl" so that it becomes the standard way of building urls in a safe way.